### PR TITLE
[Docker] Install Trition deps

### LINKED ISSUE #90377: Pytorch 2 error on provided docker but not on pip installation
## Issue description
I get an error on the model.forward(input) line  if I use a .compile model on the provided docker image: ghcr.io/pytorch/pytorch-nightly:latest
 Removing the .compile line makes the code working again. However, the very same code with .compile line works fine on a pytorch 2.0 pip installation; it fails only on the official docker image.
The 'output = model.forward(input)' line which throws the error refers to a simple resnet18 model 
'model = models.resnet18(pretrained=True)' 

## Error output:
```
root@1ab044f39dea:/workspace# python MultiImgTrain2.py --train_path CWT_train --test_path CWT_test --bs_train 256 --bs_test 1024
/opt/conda/lib/python3.10/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: /opt/conda/lib/python3.10/site-packages/torchvision/image.so: undefined symbol: _ZN5torch3jit17parseSchemaOrNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
  warn(f"Failed to load image Python extension: {e}")
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/concurrent/futures/process.py", line 246, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/codecache.py", line 488, in _worker_compile
    kernel.precompile(warm_cache_only_with_cc=cc)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/triton_ops/autotune.py", line 59, in precompile
    self.launchers = [
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/triton_ops/autotune.py", line 60, in <listcomp>
    self._precompile_config(c, warm_cache_only_with_cc)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/triton_ops/autotune.py", line 74, in _precompile_config
    triton.compile(
  File "/opt/conda/lib/python3.10/site-packages/triton/compiler.py", line 1239, in compile
    so = _build(fn.__name__, src_path, tmpdir)
  File "/opt/conda/lib/python3.10/site-packages/triton/compiler.py", line 1169, in _build
    ret = subprocess.check_call(cc_cmd)
  File "/opt/conda/lib/python3.10/subprocess.py", line 364, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/opt/conda/lib/python3.10/subprocess.py", line 345, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/opt/conda/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/conda/lib/python3.10/subprocess.py", line 1722, in _execute_child
    and os.path.dirname(executable)
  File "/opt/conda/lib/python3.10/posixpath.py", line 152, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 584, in call_user_compiler
    compiled_fn = compiler_fn(gm, self.fake_example_inputs())
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/debug_utils.py", line 915, in debug_wrapper
    compiled_gm = compiler_fn(gm, example_inputs, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/__init__.py", line 1177, in _compile_fn
    return compile_fn(model_, inputs_)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 394, in compile_fx
    return aot_autograd(
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/optimizations/training.py", line 80, in compiler_fn
    cg = aot_module_simplified(gm, example_inputs, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 2093, in aot_module_simplified
    compiled_fn = create_aot_dispatcher_function(
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 90, in time_wrapper
    r = func(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 1792, in create_aot_dispatcher_function
    compiled_fn = compiler_fn(flat_fn, fake_flat_tensor_args, aot_config)
  File "/opt/conda/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 1197, in aot_wrapper_dedupe
    return compiler_fn(flat_fn, leaf_flat_args, aot_config)
  File "/opt/conda/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 1416, in aot_dispatch_autograd
    compiled_fw_func = aot_config.fw_compiler(fw_module, flat_args_with_views_handled)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 90, in time_wrapper
    r = func(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 369, in fw_compiler
    return inner_compile(
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/debug_utils.py", line 494, in debug_wrapper
    compiled_fn = compiler_fn(gm, example_inputs, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/debug.py", line 177, in inner
    return fn(*args, **kwargs)
  File "/opt/conda/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 136, in compile_fx_inner
    compiled_fn = graph.compile_to_fn()
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/graph.py", line 500, in compile_to_fn
    return self.compile_to_module().call
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 90, in time_wrapper
    r = func(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/graph.py", line 490, in compile_to_module
    mod = PyCodeCache.load(code)
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/codecache.py", line 459, in load
    exec(code, mod.__dict__, mod.__dict__)
  File "/tmp/torchinductor_root/zg/czghpom6cfixrqvfpmhdywd2gkyfsbft3xetwhub2dznbfkkqb3e.py", line 1724, in <module>
    async_compile.wait(globals())
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/codecache.py", line 642, in wait
    scope[key] = result.result()
  File "/opt/conda/lib/python3.10/site-packages/torch/_inductor/codecache.py", line 512, in result
    self.future.result()
  File "/opt/conda/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/opt/conda/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
TypeError: expected str, bytes or os.PathLike object, not NoneType

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/workspace/MultiImgTrain2.py", line 216, in <module>
    train_model(model,train_loader)
  File "/workspace/MultiImgTrain2.py", line 131, in train_model
    output = model.forward(input)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 80, in forward
    return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 209, in _fn
    return fn(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 329, in catch_errors
    return callback(frame, cache_size)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 468, in _convert_frame
    result = inner_convert(frame, cache_size)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 102, in _fn
    return fn(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 90, in time_wrapper
    r = func(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 339, in _convert_frame_assert
    return _compile(
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 395, in _compile
    out_code = transform_code_object(code, transform)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py", line 341, in transform_code_object
    transformations(instructions, code_options)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 382, in transform
    tracer.run()
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1620, in run
    super().run()
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 484, in run
    and self.step()
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 453, in step
    getattr(self, inst.opname)(inst)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1682, in RETURN_VALUE
    self.output.compile_subgraph(self)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 439, in compile_subgraph
    self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 510, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
  File "/opt/conda/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 589, in call_user_compiler
    raise BackendCompilerFailed(self.compiler_fn, e) from e
torch._dynamo.exc.BackendCompilerFailed: _compile_fn raised TypeError: expected str, bytes or os.PathLike object, not NoneType


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True

munmap_chunk(): invalid pointer
Aborted (core dumped)
```

cc @ezyang @gchanan @zou3519 @seemethere @malfet @soumith

### PR DESCRIPTION
**Pull request by malfet**
Triton needs a working gcc, so install one from apt
Also, copy `ptxas` and `cuda.h` from conda to `/usr/local/cuda`
Add `torchaudio` to the matrix
Fix typo in workflow file

Fixes https://github.com/pytorch/pytorch/issues/90377

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -75,7 +75,12 @@ RUN case ${TARGETPLATFORM} in \
     esac && \
     /opt/conda/bin/conda clean -ya
 RUN /opt/conda/bin/pip install torchelastic
-RUN if test -n "${TRITON_VERSION}" -a "${TARGETPLATFORM}" != "linux/arm64"; then /opt/conda/bin/pip install "torchtriton==${TRITON_VERSION}" --extra-index-url https://download.pytorch.org/whl/nightly/cpu ; fi
+RUN if test -n "${TRITON_VERSION}" -a "${TARGETPLATFORM}" != "linux/arm64"; then \
+        /opt/conda/bin/pip install "torchtriton==${TRITON_VERSION}" --extra-index-url https://download.pytorch.org/whl/nightly/cpu ; \
+        apt update && apt install -y --no-install-recommends gcc; \
+        mkdir -p /usr/local/cuda/bin; cp /opt/conda/bin/ptxas /usr/local/cuda/bin; \
```
[the remark this answers is not in the document]
Comment (malfet) [reply Dockerfile:81]: Let me try to see if it will work...

[the remark this answers is not in the document]
Comment (ngimel) [reply Dockerfile:81]: For triton the var is `TRITON_PTXAS_PATH`

[the remark this answers is not in the document]
Comment (ngimel) [reply Dockerfile:81]: Yeah but `/usr/local/cuda/bin` that contains only `ptxas` is no less weird :-)

[the remark this answers is not in the document]
Comment (malfet) [reply Dockerfile:81]: `TRITION_PTXAS_PATH` only works for PTXAS, but `cuda.h` is only search in `CUDA_HOME` folder

[the remark this answers is not in the document]
Comment (malfet) [reply Dockerfile:81]: Ok, come to think of it: 
 - Short term solution should be to create `/usr/local/trition-min-cuda-X.Y` and symlink to `/usr/local/cuda/`
 - Mid-term solution is to package those dependencies into triton and make it search there if its not available in other places

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**